WordPress/WordPressApi/WordPressComOAuthClient.swift

```diff
@@ -0,0 +1,192 @@
+import AFNetworking
+
+@objc public enum WordPressComOAuthError: Int {
+    case Unknown
+    case InvalidClient
+    case UnsupportedGrantType
+    case InvalidRequest
+    case NeedsMultifactorCode
+}
+
+/**
+ `WordPressComOAuthClient` encapsulates the pattern of authenticating against WordPress.com OAuth2 service.
+
+ Right now it requires a special client id and secret, so this probably won't work for you
+
+ @see https://developer.wordpress.com/docs/oauth2/
+ */
+
+public class WordPressComOAuthClient: NSObject {
```

Comment:
Probably a good idea to make all classes `final` by default, unless we have plans to subclass it.